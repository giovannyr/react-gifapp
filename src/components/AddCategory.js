import React, { useState } from "react";
import PropTypes from 'prop-types'

export const AddCategory = ({setCategories}) => {
    const [inputValue, setInputValue] = useState('');

    const handleInputValue = (evt) => {
        setInputValue(evt.target.value);
    };

    const handleSubmit = (evt) => {
        evt.preventDefault();
        if(inputValue.trim().length > 2){
            setCategories( categories => [inputValue, ...categories]);
            setInputValue('');
        }
    };

    return (
        <form onSubmit={handleSubmit}>
            <input type="text" value={inputValue} onChange={handleInputValue} />
        </form>
    );
};

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}
