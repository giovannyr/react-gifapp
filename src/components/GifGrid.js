import React from "react";
import PropTypes from "prop-types";
import { useFetchGifs } from "../hooks/useFetchGifs";
import { GifGridItem } from "./GifGridItem";

export const GifGrid = ({ category }) => {
    const { data: images, loading } = useFetchGifs(category);

    return (
        <>
            <h3 className="animate__animated animate__flipInX animate__slow">
                {category}
            </h3>

            {loading && (
                <p className="animate__animated animate__flash animate__delay-2s">
                    cargando...
                </p>
            )}

            <div className="card-grid">
                {images.map((image) => (
                    <GifGridItem key={image.id} {...image} />
                ))}
            </div>
        </>
    );
};

GifGrid.propTypes = {
    category: PropTypes.string.isRequired,
};
