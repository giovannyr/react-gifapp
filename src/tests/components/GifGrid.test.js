import React from "react";
import "@testing-library/jest-dom";
import { shallow } from "enzyme";
import { GifGrid } from "../../components/GifGrid";
import { useFetchGifs } from "../../hooks/useFetchGifs";
jest.mock("../../hooks/useFetchGifs");

describe("pruebas en <GifGrid />", () => {
    const category = "Bart Simpson";

    test("debe mostrar componente <GifGrid /> correctamente", () => {
        useFetchGifs.mockReturnValue({
            data: [],
            loading: true,
        });

        const wrapper = shallow(<GifGrid category={category} />);
        expect(wrapper).toMatchSnapshot();
    });

    test("debe mostrar items cuando se cargan imagenes useFetchGifs", () => {
        const gifs = [
            {
                id: "abc",
                title: "Culquier titulo",
                url: "https://localhost/algo",
            },
            {
                id: "abcd",
                title: "Culquier titulo",
                url: "https://localhost/algo",
            },
        ];

        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false,
        });

        const wrapper = shallow(<GifGrid category={category} />);
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find("p").exists()).toBe(false);
        expect(wrapper.find("GifGridItem").length).toBe(gifs.length);
    });
});
