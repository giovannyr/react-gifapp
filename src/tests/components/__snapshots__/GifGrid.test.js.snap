// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en <GifGrid /> debe mostrar componente <GifGrid /> correctamente 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__flipInX animate__slow"
  >
    Bart Simpson
  </h3>
  <p
    className="animate__animated animate__flash animate__delay-2s"
  >
    cargando...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;

exports[`pruebas en <GifGrid /> debe mostrar items cuando se cargan imagenes useFetchGifs 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__flipInX animate__slow"
  >
    Bart Simpson
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="abc"
      key="abc"
      title="Culquier titulo"
      url="https://localhost/algo"
    />
    <GifGridItem
      id="abcd"
      key="abcd"
      title="Culquier titulo"
      url="https://localhost/algo"
    />
  </div>
</Fragment>
`;
